backend/app/repositories/user_repository.py

```
@@ -0,0 +1,17 @@
+class UserRepository:
+    def __init__(self, db):
+        self.collection = db.users
+
+    async def insert_user(self, user_doc):
+        try:
+            result = await self.collection.insert_one(user_doc)
+            return result
+        except Exception as e:
+            raise Exception(f"Database insert failed: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)